crates/libs/core/src/client/svc_mgmt_client.rs

```diff
@@ -150,6 +189,47 @@ impl ServiceManagementClient {
         self.remove_replica_internal(&raw, timeout.as_millis() as u32)
             .await
     }
+
+    /// Remarks:
+    /// There is a cache of service endpoints in the client that gets updated by notifications
+    /// and this same cache is used to satisfy complaint based resolution requests
+    /// (see resolve_service_partition())). Applications that both register for notifications
+    /// and use complaint based resolution on the same client instance typically only need to
+    /// pass null for the ResolvedServicePartition argument during resolution.
+    /// This will always return the endpoints in the client cache updated by the latest notification.
+    /// The notification mechanism itself will keep the client cache updated when service endpoints change.
```

Comment:
Worth documenting typical/expected latency here

Reply:
I will send another PR to compare this solution with brute force retry and added the results then.